### ISSUE TITLE
failed to select enabled extensions: failed to open database extensionData

### ISSUE THREAD

**Issue by flystaros** [hiddify/hiddify-app#1538]
### Search first

- [X] I searched and no similar issues were found

### What Happened?

I clone the code and build success
when click the connection button
there is a err dialog:
failed to select enabled extensions: failed to open database extensionData

### Minimal Reproducible Example (MRE)

1.clone code
2. flutter pub get
3. make macOS-arm64
4. flutter run
5.click connection button
6.failed to select enabled extensions: failed to open database extensionData

Device
apple m3
macOS 14.5 (23F79)
Xcode Version 16.1 (16B40)

### Expected Behavior

connect success

### Version

commit 635e30a

### Platform/OS

macOS

### Additional Context

1

### Application Config Options

```shell
1
```

### Relevant log output

```shell
024/12/11 14:54:08 Failed attempt 96 to initialize the database: stat data/extensionData.db: no such file or directory
2024/12/11 14:54:08 Failed attempt 97 to initialize the database: stat data/extensionData.db: no such file or directory
2024/12/11 14:54:08 Failed attempt 98 to initialize the database: stat data/extensionData.db: no such file or directory
2024/12/11 14:54:08 Failed attempt 99 to initialize the database: stat data/extensionData.db: no such file or directory
kk-------hiddify.go---startServices--services-: failed to select enabled extensions: failed to open database extensionDatakk---custom.go----baseDir: 0x11f80a620ERROR[0004] failed to select enabled extensions: failed to open database extensionData
Application finished.
```

### Are you willing to submit a PR? If you know how to fix the bug.

- [X] I'm willing to submit a PR (Thank you!)

**Comment by theSadra** [comment]
Hello, dear friend.
Have you been able to resolve this issue? 
Did you find any solutions for it?

**Comment by flystaros** [comment]
> Hello, dear friend. Have you been able to resolve this issue? Did you find any solutions for it? [@flystaros](https://github.com/flystaros)

yes , i resolved. i changed the libcore version and it worked. If you have any questions, feel free to contact me.

**Comment by sinais54** [comment]
Hi
When I give the sub link to HiddifyCli most of the time (99% of the time) I get the following errors and I can't connect.
.
.
.

Open http://localhost:6756/ui/?secret=SLSxqov0RLV91Bpi in your browser
2025/03/12 07:30:13 Failed attempt 0 to initialize the database: leveldb: manifest corrupted (field 'comparer'): missing [file=MANIFEST-000006]
2025/03/12 07:30:13 Failed attempt 1 to initialize the database: leveldb: manifest corrupted (field 'comparer'): missing [file=MANIFEST-000006]
2025/03/12 07:30:13 Failed attempt 2 to initialize the database: leveldb: manifest corrupted (field 'comparer'): missing [file=MANIFEST-000006]
2025/03/12 07:30:13 Failed attempt 3 to initialize the database: leveldb: manifest corrupted (field 'comparer'): missing [file=MANIFEST-000006]
2025/03/12 07:30:13 Failed attempt 4 to initialize the database: leveldb: manifest corrupted (field 'comparer'): missing [file=MANIFEST-000006]
2025/03/12 07:30:13 Failed attempt 5 to initialize the database: leveldb: manifest corrupted (field 'comparer'): missing [file=MANIFEST-000006]
2025/03/12 07:30:13 Failed attempt 6 to initialize the database: leveldb: manifest corrupted (field 'comparer'): missing [file=MANIFEST-000006]
.
.
.
Error in read and build config failed to set up global configuration: failed to open database extensionData, error: leveldb: manifest corrupted (field 'comparer'): missing [file=MANIFEST-000006]

**Comment by theSadra** [comment]
@flystaros Can you please say me that what version of libcore you used?
WIth best regards

**Comment by neomody77** [comment]
Operation log—leaving a reference for whoever comes after. . libcore is still pointing to an older version. Steps:

```
git submodule update --init --recursive   # fetch submodules
cd libcore && git checkout v3.1.8 && cd ..
make build-macos-libs
make macos-release
```
@TitanComputer Is it possible to update it in repo?
Seems to be this: 
https://github.com/hiddify/hiddify-app/pull/1386
